.github/workflows/system-tests-opensource.yml

```diff
@@ -0,0 +1,172 @@
+name: System Tests Open Source
+
+on:
+  push:
+    branches:
+      - '.+-system-tests'
+
+  schedule:
+
+    # * is a special character in YAML so you have to quote this string
+    # Run the system tests every 3 hours
+    - cron:  '0 */3 * * *'
+
+  workflow_dispatch:
+    inputs:
+      docker_registry:
+        description: 'Docker registry to pull images from (default: ghcr.io/, use registry.hub.docker.com/ for docker hub)'
+        required: true
+        default: 'ghcr.io/'
+      docker_repo:
+        description: 'Docker repo to pull images from (default: mlrun)'
+        required: true
+        default: 'mlrun'
+      test_code_from_action:
+        description: 'Take tested code from action REF (default: false - take from upstream) (note that test code will be taken from the action REF anyways)'
+        required: true
+        default: 'false'
+      ui_code_from_action:
+        description: 'Take ui code from action branch in mlrun/ui (default: false - take from upstream)'
+        required: true
+        default: 'false'
+
+env:
+  NAMESPACE: mlrun
+  MLRUN_API_NODE_PORT: 30070
+
+jobs:
+  run-system-tests-opensource-ci:
+    timeout-minutes: 60
+    name: Run System Tests
+    runs-on: ubuntu-latest
+
+    # let's not run this on every fork, change to your fork when developing
+    if: github.repository == 'mlrun/mlrun' || github.event_name == 'workflow_dispatch'
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install automation scripts dependencies and add mlrun to dev packages
```

Comment:
Leave some TODO here to try and avoid the duplication from this line until line 120